### PR TITLE
[PositionUpdater] Adapt to margin exchange 

### PR DESCRIPTION
**Pull request by Herklos**
Depends on https://github.com/Drakkar-Software/OctoBot-Trading/pull/103

### CODE REVIEW

File: octobot_trading/constants.py [hunk 1]
```diff
@@ -51,6 +51,7 @@
 CONFIG_EXCHANGE_KEY = "api-key"
 CONFIG_EXCHANGE_SECRET = "api-secret"
 CONFIG_EXCHANGE_PASSWORD = "api-password"
+CONFIG_EXCHANGE_SANDBOXED = "sandboxed"
```
Comment (GuillaumeDSM) [inline octobot_trading/constants.py]: 👍

File: octobot_trading/exchanges/exchange_builder.py [hunk 2]
```diff
@@ -94,7 +91,7 @@ def is_backtesting(self, backtesting_files):
         return self
 
     def is_sandboxed(self, sandboxed: bool):
-        self._is_exchange_manager_sandboxed = sandboxed
+        self.config[CONFIG_EXCHANGES][self.exchange_name][CONFIG_EXCHANGE_SANDBOXED] = sandboxed
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/exchange_builder.py]: Not sure it's a good idea to edit the user config

File: octobot_trading/exchanges/margin/margin_exchange.py [hunk 3]
```diff
@@ -53,3 +53,6 @@ def get_name(cls) -> str:
 
     async def get_position_status(self, position_id, symbol=None):
         raise NotImplementedError("get_position_status is not implemented")
+
+    def cleanup_positions_dict(self, positions):
+        return positions
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/margin/margin_exchange.py]: missing newline

File: octobot_trading/exchanges/rest_exchange.py [hunk 4]
```diff
@@ -325,17 +328,23 @@ def get_uniform_timestamp(self, timestamp):
     def get_pair_from_exchange(self, pair) -> str:
         try:
             return self.client.market(pair)["symbol"]
-        except ccxt.base.errors.BadSymbol:
-            self.logger.error(f"Failed to get market of {pair}")
-            return None
+        except BadSymbol:
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/rest_exchange.py]: No import required ?

File: octobot_trading/exchanges/exchange_builder.py (hunk 2 above)
Comment (Herklos) [reply octobot_trading/exchanges/exchange_builder.py]: I agree, but it's a way of letting users set the sandbox mode from config and also forcing sandbox mode from bot params even if it's not set in user config.

File: octobot_trading/exchanges/rest_exchange.py (hunk 4 above)
Comment (Herklos) [reply octobot_trading/exchanges/rest_exchange.py]: Already done [here](https://github.com/Drakkar-Software/OctoBot-Trading/pull/106/files/17ca52dbe0bc55b7b3c392b2301d8267127aa106..60d93597a36382259a99680f1cd578bce24aec71#diff-29e2db0b0f5a93571a9f481b0c2fe4a9R21)

Comment (GuillaumeDSM) [reply octobot_trading/exchanges/rest_exchange.py]: Oh, right, I didn't see it

File: octobot_trading/exchanges/exchange_builder.py (hunk 2 above)
Comment (GuillaumeDSM) [reply octobot_trading/exchanges/exchange_builder.py]: Right, in this case shouldn't we use something like:
```python
self._is_exchange_manager_sandboxed = self.config[CONFIG_EXCHANGES][self.exchange_name][CONFIG_EXCHANGE_SANDBOXED] or sandboxed
```

File: octobot_trading/exchanges/exchange_builder.py [hunk 5]
```diff
@@ -94,7 +90,8 @@ def is_backtesting(self, backtesting_files):
         return self
 
     def is_sandboxed(self, sandboxed: bool):
-        self._is_exchange_manager_sandboxed = sandboxed
+        self.exchange_manager.is_sandboxed = sandboxed or \
+                                             self.config[CONFIG_EXCHANGES][self.exchange_name][CONFIG_EXCHANGE_SANDBOXED]
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/exchange_builder.py]: 👍

File: octobot_trading/exchanges/margin/margin_exchange.py (hunk 3 above)
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/margin/margin_exchange.py]: WIP ?

File: octobot_trading/exchanges/rest_exchange.py [hunk 6]
```diff
@@ -26,7 +26,7 @@
 from octobot_commons.enums import TimeFramesMinutes
 from octobot_trading.constants import CONFIG_EXCHANGES, CONFIG_EXCHANGE_KEY, CONFIG_EXCHANGE_SECRET, \
     CONFIG_EXCHANGE_PASSWORD, CONFIG_DEFAULT_FEES, CONFIG_PORTFOLIO_INFO, CONFIG_PORTFOLIO_FREE, CONFIG_PORTFOLIO_USED, \
-    CONFIG_PORTFOLIO_TOTAL
+    CONFIG_PORTFOLIO_TOTAL, CONFIG_EXCHANGE_SANDBOXED
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/rest_exchange.py]: is ```CONFIG_EXCHANGE_SANDBOXED``` used ?

File: octobot_trading/exchanges/margin/margin_exchange.py (hunk 3 above)
Comment (Herklos) [reply octobot_trading/exchanges/margin/margin_exchange.py]: No it's the default behaviour. For example it's overwritten in bitmex MarginExchange tentacle.

**Pull request merged**